core/src/main/java/com/redhat/lightblue/client/LightblueClient.java

```diff
@@ -4,14 +4,15 @@
 
 import com.redhat.lightblue.client.request.AbstractLightblueDataRequest;
 import com.redhat.lightblue.client.request.AbstractLightblueMetadataRequest;
+import com.redhat.lightblue.client.request.LightblueRequest;
 import com.redhat.lightblue.client.response.LightblueResponse;
 
 public interface LightblueClient {
 
-    public abstract LightblueResponse metadata(AbstractLightblueMetadataRequest lightblueRequest);
+    LightblueResponse metadata(LightblueRequest lightblueRequest);
 
-    public abstract LightblueResponse data(AbstractLightblueDataRequest lightblueRequest);
+    LightblueResponse data(LightblueRequest lightblueRequest);
 
-    public abstract <T> T data(AbstractLightblueDataRequest lightblueRequest, Class<T> type) throws IOException;
+    <T> T data(AbstractLightblueDataRequest lightblueRequest, Class<T> type) throws IOException;
```

Comment:
I see what you are doing and I am fine with switching this, but I would like to see LightblueHttpDataRequest and LightblueHttpMetadataRequest throw an exception instead of returning a null value.